src/main/java/com/linecorp/armeria/server/metrics/MetricsService.java

```diff
@@ -0,0 +1,32 @@
+package com.linecorp.armeria.server.metrics;
+
+import java.util.function.Function;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linecorp.armeria.server.Service;
+
+/**
+ * A {@link Service} decorator that tracks request stats using the Dropwizard metrics library.
+ * To use, simply prepare a {@link MetricRegistry} and add this decorator to a service specification.
+ *
+ * <p>Example:
+ * <pre>{@code
+ * MetricRegistry metricRegistry = new MetricRegistry();
+ * serverBuilder.serviceAt("/service", ThriftService.of(handler)
+ *     .decorate(MetricsService.newDecorator("MyService", metricRegistry)));
+ * }
+ * </pre>
+ * </p>
+ * <p>It is generally recommended to define your own name for the service instead of using something like
+ * the Java class to make sure otherwise safe changes like renames don't break metrics.</p>
+ */
+public final class MetricsService {
```

Comment:
I think method in this class can be merged into `MetricCollectingService`, for reducing naming ambiguity.